src/AppBundle/Entity/Receipt.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity(repositoryClass="AppBundle\Entity\Repository\ReceiptRepository")
+ * @ORM\Table(name="receipt")
+ */
+class Receipt
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="User", inversedBy="receipts", cascade={"persist"})
+     */
+    private $user;
+
+    /**
+     * @ORM\Column(type="datetime", nullable=true)
+     */
+    private $submitDate;
+
+    /**
+     * @ORM\Column(name="picture_path", type="string", length=45, nullable=true)
+     */
+    private $picturePath;
+
+    /**
+     * @ORM\Column(type="string")
+     * @Assert\NotBlank(message="Dette feltet kan ikke være tomt.")
+     */
+    private $description;
+
+    /**
+     * @ORM\Column(type="float")
+     * @Assert\NotBlank(message="Dette feltet kan ikke være tomt.")
+     */
+    private $sum;
+
+    /**
+     * @var bool
+     * @ORM\Column(type="boolean")
+     */
+    private $active;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $visualId;
+
+    /**
+     * Receipt constructor.
+     */
+    public function __construct()
+    {
+        $this->active = true;
+        $this->submitDate = new \DateTime();
+        $this->visualId = substr(md5(uniqid()), -7);
```

Comment:
Det ser ut som det ble mye kompleksitet for å gjøre `visualId` unik, spesielt når den er avhengig av ID som ikke finnes når objektet opprettes.

Her lukter det litt "Premature Generalization", lage funksjonalitet og introdusere kompleksitet som man kanskje (sannsynligvis ikke) får bruk for i fremtiden:
> This is really going to be a clean framework. I'll make an abstract class out of this part so that folks can subclass it later, and I'll put in a bunch of well-commented overridable hooks in the concrete subclasses so that folks can use them as templates, and just in case somebody ever needs to build special debug subclasses, I'll put in extra stubs over there (somebody will thank me for 'em one of these days). Yeah, this is really going to be neat.
>
>Thus is bloated software produced when our artistic sense gets the better of us. -- DaveSmith
http://wiki.c2.com/?PrematureGeneralization

Hvis vi trenger unike IDer senere kan vi skrive et lite script som looper over alle eksisterende kvitteringer og oppretter nye IDer.

Høres det fornuftig ut om vi beholder forrige løsning?

Her er et annet alternativ som gir unike IDer med mindre to kvitteringer blir opprettet på nøyaktig samme millisekund:
```php
$currentTimeInMilliseconds = round(microtime(true) * 1000);
$visualId = dechex($currentTimeInMilliseconds);
```